WordPress/gutenberg
# [RNMobile] Add block transforms integration tests for Text blocks

## What?


Add integration tests to cover the block transform functionality of Text blocks.

## Why?


This PR is a follow-up of https://github.com/WordPress/gutenberg/pull/48792 to cover with tests the block transform functionality.

## How?


Integration tests have been added to the following Text blocks:
- Paragraph block
- Heading block
- List block
- Quote block
- Preformatted block
- Shortcode block
- Verse block
- Pullquote block

Additionally, two helpers have been added to simplify the test logic:
- `transformBlock`: Transforms a block using the Block actions and Transform menus.
- `getBlockTransformOptions`: Returns the transform options. This is used to check that the block transform options of a block match the expected ones.

## Testing Instructions




### Unit tests pass
1. Check that the `Unit Tests / Mobile` job pass.

### Manual test
Follow the change introduced in https://github.com/WordPress/gutenberg/pull/48792 that enables more block transforms, it would be great to test the transformations manually to confirm that work as expected.

1. Open the editor in the browser and the app.
2. Open the transformation menu on both platforms and observe that match the same options.
**NOTE:** The web version usually displays more options because it supports more blocks. We should only check [the supported blocks in the native version](https://github.com/WordPress/gutenberg/blob/54d43c40da5a1e2a7e86f4dfe41280d64ef5f3ca/packages/block-library/src/index.native.js#L68-L114).
3. Perform the block transformations and compare the results between platforms (**check the collapsed list below**).

<details><summary><strong>Here you can check the potential block transformations for the blocks</strong></summary>

## Paragraph
- Heading 
- List 
- Quote 
- Buttons ❌: This transformation is only supported in the web version, as in native it needs `document.createHTMLDocument` function to work.
- Preformatted 
- Columns 
- Verse 
- Pullquote 
- Group 

##  Heading
- Paragraph 
- List 
- Quote 
- Columns 
- Pullquote 
- Group 

## List
- Paragraph 
- Heading 
- Quote 
- Columns 
- Group 

##  Quote
- Unwrap 
- Columns 
- Group 
- Pullquote (if only text)

##  Preformatted
- Paragraph 
- Columns 
- Group 

##  Shortcode
- Columns 
- Groups 

##  Verse
- Paragraph 
- Columns 
- Group 

##  Pullquote
- Paragraph 
- Heading 
- Quote 
- Columns 
- Group 

</details>

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A
